transport/swift/swift.go

```diff
@@ -0,0 +1,101 @@
+package swift
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/imgproxy/imgproxy/v3/config"
+	"github.com/ncw/swift/v2"
+)
+
+type transport struct {
+	con *swift.Connection
+}
+
+func New() (http.RoundTripper, error) {
+	c := &swift.Connection{
+		UserName:       config.SwiftUsername,
+		ApiKey:         config.SwiftAPIKey,
+		AuthUrl:        config.SwiftAuthURL,
+		AuthVersion:    config.SwiftAuthVersion,
+		Domain:         config.SwiftDomain, // v3 auth only
+		Tenant:         config.SwiftTenant, // v2 auth only
+		Timeout:        time.Duration(config.SwiftTimeoutSeconds) * time.Second,
+		ConnectTimeout: time.Duration(config.SwiftConnectTimeoutSeconds) * time.Second,
+	}
+
+	ctx := context.Background()
+
+	err := c.Authenticate(ctx)
+
+	if err != nil {
+		return nil, fmt.Errorf("swift authentication error: %s", err)
+	}
+
+	return transport{con: c}, nil
+}
+
+func (t transport) RoundTrip(req *http.Request) (resp *http.Response, err error) {
+	if err != nil {
+		return &http.Response{
+			Status:     "400 Bad Request",
+			StatusCode: 400,
+			Proto:      "HTTP/1.1",
+			ProtoMajor: 1,
+			ProtoMinor: 1,
+			Close:      true,
+			Request:    req,
+		}, err
+	}
```

Comment:
This is useless now as we don't have an error to check

Reply:
🤦 Nice catch. I forgot to clean that up. It's done now.